### ISSUE TITLE
False positive: accessing variables from other, possibly undefined modules

### ISSUE THREAD

**Issue by mbaz** [aviatesk/JET.jl#516]
Consider code whose behavior depends on whether it is running on a notebook such as IJulia. Such a test is usually performed by inspecting a variable in the notebook's module. For example, [IJulia's documentation recommends](https://julialang.github.io/IJulia.jl/stable/manual/usage/#Detecting-that-code-is-running-under-IJulia) this code to test whether we're running in a notebook:
```
isdefined(Main, :IJulia) && Main.IJulia.inited
```
JET detects `IJulia` as not defined:
```
  ═════ 1 possible error found ═════
  ┌ @ /home/miguel/rcs/jdev/Gaston/src/Gaston.jl:87 Main.IJulia
  │┌ @ Base.jl:31 Base.getglobal(x, f)
  ││ `IJulia` is not defined
  │└──────────────
```
However, in practice this will never be a problem, since the `IJulia` module is only accessed if it is defined.

**Comment by aviatesk** [comment]
Yes, this false positive is very annoying. I will try to fix it sometime in this week.
